### PR TITLE
DOC: Add ``np.polynomial.laguerre`` examples

### PR DESCRIPTION
**Pull request by linus-md**
This adds examples for several functions in `np.polynomial.laguerre`.

### CODE REVIEW

File: numpy/polynomial/laguerre.py [hunk 1]
```diff
@@ -1595,6 +1657,13 @@ def lagweight(x):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial.laguerre import lagweight
+    >>> x = np.array([0, 1, 2])
+    >>> lagweight()
```
Comment (eendebakpt) [inline numpy/polynomial/laguerre.py:1664]:
```suggestion
    >>> lagweight(x)
```
Missing argument to `lagweight`. The CI tests pass, so this means there are no doctests running. That might be something to consider.

File: numpy/polynomial/laguerre.py [hunk 2]
```diff
@@ -1098,6 +1118,17 @@ def laggrid3d(x, y, z, c):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial.laguerre import laggrid3d
+    >>> c = [[[1, 2],[3, 4]],[[5, 6],[7, 8]]]
```
Comment (eendebakpt) [inline numpy/polynomial/laguerre.py:1124]:
```suggestion
    >>> c = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]]
```
pep8 rules

File: numpy/polynomial/laguerre.py [hunk 3]
```diff
@@ -1042,6 +1055,13 @@ def lagval3d(x, y, z, c):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial.laguerre import lagval3d
+    >>> c = [[[1, 2],[3, 4]],[[5, 6],[7, 8]]]
```
Comment (eendebakpt) [inline numpy/polynomial/laguerre.py:1061]:
```suggestion
    >>> c = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]]
```

File: numpy/polynomial/laguerre.py [hunk 4]
```diff
@@ -991,6 +996,14 @@ def laggrid2d(x, y, c):
 
     .. versionadded:: 1.7.0
 
+    Examples
+    --------
+    >>> from numpy.polynomial.laguerre import laggrid2d
+    >>> c = [[1, 2],[3, 4]]
```
Comment (eendebakpt) [inline numpy/polynomial/laguerre.py:1002]:
```suggestion
    >>> c = [[1, 2], [3, 4]]
```

File: numpy/polynomial/laguerre.py (hunk 1 above)
Comment (linus-md) [reply numpy/polynomial/laguerre.py]: You are right! This might be something to look at for the maintainers?

Comment (mtsokol) [reply numpy/polynomial/laguerre.py]: I managed to reproduce it locally. 
It looks like `PUBLIC_SUBMODULES` in `refguide_check.py` is missing a bunch of submodules [link](https://github.com/numpy/numpy/blob/35c4319f33f3eafa091d4c9790b6de2468874b51/tools/refguide_check.py#L62C6-L62C6) (it doesn't discover them recursively, they need to be listed explicitly, just as [here](https://github.com/numpy/numpy/blob/35c4319f33f3eafa091d4c9790b6de2468874b51/numpy/tests/test_public_api.py#L110)).

**Review by mtsokol — APPROVED** [review]

File: numpy/polynomial/laguerre.py (hunk 1 above)
Comment (mtsokol) [reply numpy/polynomial/laguerre.py]: Opened a PR here: https://github.com/numpy/numpy/pull/25367

**Comment by mtsokol** [comment]
Last thing - please squash PR into one commit with `DOC: ...` message.

**Review by rgommers — APPROVED** [review]
LGTM too, thanks all!

**Comment by rgommers** [comment]
Windows CI failure is unrelated, that's a Chocolatey connection issue.

**Pull request merged**